cooltools/sandbox/pixel_filter_util.py

```diff
@@ -0,0 +1,214 @@
+    """
+    if isinstance(clr, float):
+        raise TypeError(
+            "If only threshold value is provided, please use 'thres' keyword to set threshold value (e.g. thres=0.2)"
+        )
+    coverage = cooltools.coverage(clr)
+    cis_total_cov = coverage[0] / coverage[1]
+    bin_mask = cis_total_cov > thres
+
+    return bin_mask
+
+
+def generate_bin_mask(
+    clr, filters=[None], store=False, store_name="cis_total_ratio_>_0.5_thres"
+):
+    """
+    Generates a binary mask for a given `clr` object based on a list of filters and thresholds.
+
+    Parameters
+    ----------
+    clr : cooler.Cooler
+        A cooler object containing Hi-C contact matrices.
+    filters : list
+        A list of filter functions to apply to the contact matrices.
+    store : bool, optional
+        If True, store the results in the input cooler file when finished. Default is False.
+    store_name : str, optional
+        Name of the columns of the bin table to save bin mask.
+
+    Returns
+    -------
+    bin_mask : numpy.ndarray
+        A binary mask indicating which genomic bins pass all filters.
+    """
+    if not isinstance(filters, list):
+        logger.error("filter_lst parameter takes a list")
+
+    bin_mask = np.array([True] * clr.bins().shape[0])
+    for filter in filters:
+        bin_mask *= filter(clr)
+
+    if store:
+        with clr.open("r+") as grp:
+            if store_name in grp["bins"]:
+                del grp["bins"][store_name]
+            h5opts = dict(compression="gzip", compression_opts=6)
+            grp["bins"].create_dataset(store_name, data=bin_mask, **h5opts, dtype=bool)
+
+    return bin_mask
+
+
+def _pixel_filter(chunk_pixels, good_bins_index):
+    """
+    Filters a chunk of pixels based on a list of good bin indices.
+
+    Parameters
+    ----------
+    chunk_pixels : pandas.DataFrame
+        A DataFrame containing the pixels to be filtered. It must have columns 'bin1_id' and 'bin2_id'.
+    good_bins_index : list of int
+        A list of indices representing the good bins.
+
+    Returns
+    -------
+    pandas.DataFrame
+        A DataFrame containing only the pixels whose bin1_id and bin2_id are in good_bins_index.
+    """
+
+    pixels_mask = chunk_pixels["bin1_id"].isin(good_bins_index) * chunk_pixels[
+        "bin2_id"
+    ].isin(good_bins_index)
+    return chunk_pixels[pixels_mask]
+
+
+def pixel_iter_chunks(clr, chunksize):
+    """
+    Iterate over the pixels of a cooler object in chunks of a given size.
+
+    Parameters
+    ----------
+    clr : cooler.Cooler
+        A cooler object containing Hi-C data.
+    chunksize : int
+        The size of each chunk of pixels to iterate over.
+
+    Yields
+    ------
+    chunk : numpy.ndarray
+        A chunk of pixels of size `chunksize`.
+    """
+    selector = clr.pixels()
+    for lo, hi in cooler.util.partition(0, len(selector), chunksize):
+        chunk = selector[lo:hi]
+        yield chunk
+
+
+def pool_decorator(func):
```

Comment:
Can probably delete this one & import from lib?

Reply:
updated